core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -378,6 +378,11 @@ sealed abstract class IO[E, A] { self =>
    */
   final def retry: IO[E, A] = self orElse retry
 
+  final def retryIf(p: E => Boolean): IO[E, A] =
+    self.catchSome {
+      case e: Any if p(e.asInstanceOf[E]) => retryIf(p)
```

Comment:
Why the need for this unsafe cast?

Reply:
I cannot `case e: E ...` due to type erasure

Reply:
It should already be `E`, i.e.:

```scala
self.catchSome {
  case e if p(e) => retryIf(p)
}
```

Reply:
Of course! silly me.